agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -72,13 +73,13 @@ class AtumContext private[agent] (
     this
   }
 
-  def createCheckpointOnProvidedData(checkpointName: String, author: String, measurements: Seq[Measurement]): AtumContext = {
+  def createCheckpointOnProvidedData(checkpointName: String, measurements: Seq[Measurement]): AtumContext = {
     val offsetDateTimeNow = OffsetDateTime.now()
 
     val checkpointDTO = CheckpointDTO(
```

Comment:
Hmm, like a companion object for this DTO that would have an apply method?

Reply:
Actually you are right, that would be a wrong direction dependency.. Maybe instead a method in Portioning to create the DTO? The thing is, DTO is created twice the same way - DRY.

Reply:
Yep exactly. I wanted to have DTOs that are a simple case classes, no logic or so around them. Okay aI'll think about it :)

Reply:
Actually...is it really worth it? It's on 2 places that are almost the same, yes, but is it such a big issue? It will probably not grow.  

Such method / function you propose would have tons of parameters and I don't think it would really make the code better. If you insist I can, but I'm just not convinced to create auxiliary functions when the internals of such functions are so trivial & when it's not reusable on more places

Reply:
Would be nice, but not a priority now. 👍